### ISSUE TITLE
Carthage may not be updating checkouts for transitive dependencies

### ISSUE THREAD

**Issue by sharplet** [Carthage/Carthage#2349]
Because I use the prebuilt binaries, my workflow when adding a new dependency is usually to run `carthage update NewDependency` to avoid rebuilding unrelated/unchanged libraries. When I ran this command, it looks like the transitive dependency wasn't checked out, wasn't built, and curiously when I tried to build it directly with `carthage build`, just did nothing.

* `carthage version`: 0.28.0
* `xcodebuild -version`: 9.2 (Build version 9C40b)
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? Yes
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No

**Cartfile**
```diff
 github "matthewpalmer/Locksmith"
+github "thoughtbot/Perform"
 github "Thomvis/BrightFutures"
```

**Carthage Output**
```
% carthage update --platform iOS --no-use-binaries Perform
*** Fetching GCDWebServer
*** Fetching OHHTTPStubs
*** Fetching Nimble
*** Fetching Perform
*** Fetching Locksmith
*** Fetching BrightFutures
*** Fetching Aspects
*** Fetching Result
*** Checking out Perform at "2.0.0"
*** xcodebuild output can be found in /var/folders/nl/09nb5jwj5934y_z1cl7wbxg00000gn/T/carthage-xcodebuild.hFJkGb.log
*** Building scheme "Perform" in Perform.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project <project-path>/Carthage/Checkouts/Perform/Perform.xcodeproj -scheme Perform -configuration Release -derivedDataPath <home>/Library/Caches/org.carthage.CarthageKit/DerivedData/9.2_9C40b/Perform/2.0.0 -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/nl/09nb5jwj5934y_z1cl7wbxg00000gn/T/Perform SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO (launched in <home>/src/peerfit/pf-ios/Carthage/Checkouts/Perform)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/nl/09nb5jwj5934y_z1cl7wbxg00000gn/T/carthage-xcodebuild.hFJkGb.log
```

**Cartfile.resolved**
```diff
 github "AliSoftware/OHHTTPStubs" "6.1.0"
 github "Quick/Nimble" "v7.0.3"
 github "Thomvis/BrightFutures" "6.0.0"
 github "antitypical/Result" "3.2.4"
 github "matthewpalmer/Locksmith" "4.0.0"
+github "steipete/Aspects" "1.4.2"
 github "swisspol/GCDWebServer" "3.4.2"
+github "thoughtbot/Perform" "2.0.0"
```

**Comment by mdiep** [comment]
This may address it, but I haven't had time to review yet: https://github.com/Carthage/Carthage/pull/2321

**Comment by mdiep** [comment]
Which means this is probably a duplicate of #2188?

**Comment by blender** [comment]
Duplicate of #2188

**Comment by blender** [comment]
Fixed in 0.29.0
